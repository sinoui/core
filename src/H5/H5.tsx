import { TextContent, typographyPub } from '../utils/typographyPub';

const H5 = typographyPub('h5', 'h5')(TextContent);
export default H5;
