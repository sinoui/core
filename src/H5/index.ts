import createTypography from '../Typography/createTypography';

export default createTypography('h5', 'H5');
