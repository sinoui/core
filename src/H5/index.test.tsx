import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import H5 from './index';

/**
 * H5组件单元测试
 */
describe('文字排版 单元测试', () => {
  afterEach(cleanup);

  it('指定不同的根元素', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <H5 as="div">这是一个文字。</H5>
      </ThemeProvider>,
    );

    const text = container.querySelector('div');
    expect(text).toHaveTextContent('这是一个文字。');
  });
  it('通过 gutterBottom 指定下边距', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <H5 gutterBottom>这是一个文字。</H5>
      </ThemeProvider>,
    );

    const text = container.querySelector('.sinoui-typography--gutter-bottom');
    expect(text).toHaveTextContent('这是一个文字。');
  });
  it('ref 指向根元素', () => {
    const ref: React.RefObject<HTMLInputElement> = React.createRef();
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <H5 data-testid="typo" ref={ref}>
          这是一个文字。
        </H5>
      </ThemeProvider>,
    );

    const text = container.firstChild;
    expect(text).toEqual(ref.current);
  });
  it('设置align文本对齐方式', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <H5 data-testid="typo" align="center">
          这是一个文字。
        </H5>
      </ThemeProvider>,
    );

    const text = container.firstChild;
    expect(text).toHaveStyle(`text-align:center`);
  });
});

/**
 * H5组件快照测试
 */
describe('文字排版 快照测试', () => {
  it('基本文字排版', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <H5>H5</H5>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('设置属性', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <H5 as="h2">H5</H5>
          <H5 align="center">H5</H5>
          <H5 noWrap>H5</H5>
          <H5 color="success">H5</H5>
          <H5 gutterBottom>H5</H5>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('颜色设置', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <H5>H5</H5>
          <H5 color="primary">H5</H5>
          <H5 color="secondary">H5</H5>
          <H5 color="textPrimary">H5</H5>
          <H5 color="textSecondary">H5</H5>
          <H5 color="error">H5</H5>
          <H5 color="warning">H5</H5>
          <H5 color="success">H5</H5>
          <H5 color="info">H5</H5>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
