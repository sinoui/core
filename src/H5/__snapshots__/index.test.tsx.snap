// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`文字排版 快照测试 基本文字排版 1`] = `
<div
  className="TestWrapper__Container-sc-1mof8la-0 fadnqN"
>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 ykqwj sinoui-typography sinoui-typography--h5"
    color="textPrimary"
    type="h5"
  >
    H5
  </h5>
</div>
`;

exports[`文字排版 快照测试 设置属性 1`] = `
<div
  className="TestWrapper__Container-sc-1mof8la-0 fadnqN"
>
  <h2
    className="Typography__TypographyWrapper-sc-12agh80-0 ykqwj sinoui-typography sinoui-typography--h5"
    color="textPrimary"
    type="h5"
  >
    H5
  </h2>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 jiHqCd sinoui-typography sinoui-typography--h5"
    color="textPrimary"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 bCywnC sinoui-typography sinoui-typography--h5"
    color="textPrimary"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 djbPTi sinoui-typography sinoui-typography--h5"
    color="success"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 dFYCSw sinoui-typography sinoui-typography--h5 sinoui-typography--gutter-bottom"
    color="textPrimary"
    type="h5"
  >
    H5
  </h5>
</div>
`;

exports[`文字排版 快照测试 颜色设置 1`] = `
<div
  className="TestWrapper__Container-sc-1mof8la-0 fadnqN"
>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 ykqwj sinoui-typography sinoui-typography--h5"
    color="textPrimary"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 kBooeQ sinoui-typography sinoui-typography--h5"
    color="primary"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 bcCbkN sinoui-typography sinoui-typography--h5"
    color="secondary"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 ykqwj sinoui-typography sinoui-typography--h5"
    color="textPrimary"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 CPuDs sinoui-typography sinoui-typography--h5"
    color="textSecondary"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 gIpGEK sinoui-typography sinoui-typography--h5"
    color="error"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 iLWGjp sinoui-typography sinoui-typography--h5"
    color="warning"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 djbPTi sinoui-typography sinoui-typography--h5"
    color="success"
    type="h5"
  >
    H5
  </h5>
  <h5
    className="Typography__TypographyWrapper-sc-12agh80-0 dihnvQ sinoui-typography sinoui-typography--h5"
    color="info"
    type="h5"
  >
    H5
  </h5>
</div>
`;
